--- patches/source/freetype/freetype.SlackBuild	2015-01-14 03:48:55.000000000 +0530
+++ patches/source/freetype/freetype.SlackBuild.new	2015-01-18 16:51:33.282633938 +0530
@@ -20,6 +20,7 @@
 #  OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
 #  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
+set -e
 
 PKGNAM=freetype
 VERSION=${VERSION:-$(echo $PKGNAM-*.tar.xz | cut -d - -f 2 | rev | cut -f 3- -d . | rev)}
@@ -80,6 +81,15 @@
 #   http://www.freetype.org/patents.html
 #zcat $CWD/freetype.subpixel.rendering.diff.gz | patch -p1 --verbose || exit 1
 
+# infinality-bundle
+INFINALITY_BUNDLE=$CWD/infinality-bundle
+PATCHES="01-freetype-2.5.5-enable-valid.patch 02-ftsmooth-2.5.5.patch 04-infinality-2.5.5-2014.12.30.patch"
+for patch in $PATCHES; do
+  echo "Applying:"
+  echo "$INFINALITY_BUNDLE/$patch ..."
+  patch -p1 < $INFINALITY_BUNDLE/$patch
+done
+
 chown -R root:root .
 CFLAGS="$SLKCFLAGS" make setup CFG="--prefix=/usr --libdir=/usr/lib${LIBDIRSUFFIX} --build=$ARCH-slackware-linux"
 make $NUMJOBS || make || exit 1
@@ -112,9 +122,13 @@
 # too much
 rm -rf $PKG/usr/doc/freetype-$VERSION/docs/reference
 
+# infinality-bundle
+install -D -m0755 $INFINALITY_BUNDLE/generic_settings/infinality-settings.sh \
+  $PKG/etc/profile.d/infinality-settings.sh
+
 mkdir $PKG/install
 cat $CWD/slack-desc > $PKG/install/slack-desc
 
 cd $PKG
-/sbin/makepkg -l y -c n $TMP/freetype-$VERSION-$ARCH-$BUILD.txz
+/sbin/makepkg -l y -c n $TMP/freetype-"$VERSION"_infinality-$ARCH-$BUILD.txz
 
